### DISCUSSION TITLE
Can't toggle by using '@click' with onClickOutside

### DISCUSSION THREAD

**Discussion by anburocky3** [vueuse/vueuse#2181] [Q&A]
I can understand the usage of `onClickOutside` it is an good helper for modal dialogs, but i have trouble in implementing the menu dropdown.

**The problem:**
If I click "Toggle Modal" button, it is opening, but it is not toggling. I think, `v-on-click-outside` is interrupting in some way. Is there any way to make this toggle work in `v-on-click-outside` directive?

```html
<script setup lang="ts">
import { ref } from 'vue'
import { vOnClickOutside } from '@vueuse/components'

const modal = ref(false)
function closeModal() {
  modal.value = false
}

</script>

<template>
  <button @click="modal = !modal">
    Toggle Modal
  </button>
  <div v-if="modal" v-on-click-outside="closeModal">
    Hello World
  </div>
</template>
```

**Similar issues:**

- https://github.com/vueuse/vueuse/issues/1405 (But its too technical, any short example like the above would help)

**Comment by sibbng** [comment]
I made a PR to add this feature: https://github.com/vueuse/vueuse/pull/2183

**Reply by anburocky3** [reply]
Can't wait to try those. <3 Thank you.
